packages/synapse-react-client/src/components/ChallengeRegisterButton/ChallengeRegisterButton.tsx

```diff
@@ -1,102 +1,87 @@
-import React, { useEffect, useState } from 'react'
+import React, { useEffect } from 'react'
 import ExitToAppIcon from '@mui/icons-material/ExitToApp'
-import { Box } from '@mui/system'
+import { Box } from '@mui/material'
 import SpinnerButton from '../SpinnerButton/SpinnerButton'
-import { useSynapseContext } from '../../utils'
 import {
   useGetCurrentUserProfile,
   useGetEntityChallenge,
+  useGetIsUserMemberOfTeam,
   useGetUserSubmissionTeams,
 } from '../../synapse-queries'
-import { Challenge, PaginatedIds } from '@sage-bionetworks/synapse-types'
-import { useGetIsUserMemberOfTeam } from '../../synapse-queries/team/useTeamMembers'
-import { SynapseClientError } from '../../utils/SynapseClientError'
+import { SynapseClientError, useSynapseContext } from '../../utils'
 
 export interface ChallengeRegisterButtonProps {
   projectId: string
-  onChallengeError?: (error: SynapseClientError) => void
+  onError?: (error: SynapseClientError) => void
   onJoinClick?: () => void
   onLeaveClick?: () => void
 }
 
-const EMPTY_ID = ''
-
 const ChallengeRegisterButton = ({
   projectId,
-  onChallengeError,
+  onError,
   onJoinClick,
   onLeaveClick,
 }: ChallengeRegisterButtonProps) => {
   const { accessToken } = useSynapseContext()
-  const [challenge, setChallenge] = useState<Challenge>()
-  const { data: userProfile } = useGetCurrentUserProfile()
-  const [isRegistered, setIsRegistered] = useState<boolean>(false)
-  const [hasSubmissionTeam, setHasSubmissionTeam] = useState<boolean>(false)
-  const [loading, setLoading] = useState<boolean>(true)
-  const [requestError, setRequestError] = useState<SynapseClientError>()
-
-  useEffect(() => {
-    if (requestError && onChallengeError) onChallengeError(requestError)
-  }, [requestError, onChallengeError])
+  const isLoggedIn = !!accessToken
+  const { data: userProfile } = useGetCurrentUserProfile({
+    enabled: isLoggedIn,
+  })
 
-  useGetEntityChallenge(projectId, {
-    enabled: !!accessToken && !challenge,
-    onSettled: (data, error) => {
-      if (data) {
-        setChallenge(data)
-      }
-      if (error) {
-        setLoading(false)
-        setRequestError(error)
-      }
-    },
```

Comment:
These `onSettled` callbacks are removed in @tanstack/react-query v5, so remove them here.

Using `onSettled` here could cause issues where these state variables are not properly updated. [More information here](https://github.com/TanStack/query/discussions/5279).